pkg/cloud/services/ec2/instances.go

```diff
@@ -355,6 +361,10 @@ func (s *Service) getFilteredSubnets(criteria ...*ec2.Filter) ([]*ec2.Subnet, er
 // GetCoreSecurityGroups looks up the security group IDs managed by this actuator
 // They are considered "core" to its proper functioning
 func (s *Service) GetCoreSecurityGroups(scope *scope.MachineScope) ([]string, error) {
+	if scope.IsExternallyManaged() {
+		return nil, nil
+	}
```

Comment:
This seems to have the same effect as the changes on lines 191 right?

Also, do we not still need security groups? Or are they just fetched explicitly from the AWSCluster in this case?

Reply:
>This seems to have the same effect as the changes on lines 191 right?
I think this is called in more places, so we possibly want to drop the other one and keep this logic here within the func.

>Also, do we not still need security groups? Or are they just fetched explicitly from the AWSCluster in this case?

I expect machines with externallyManaged infra to set their desired security groups explicitly through the API contract https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1alpha4/awsmachine_types.go#L100
and don't do any infra naming convention assumptions which would be driven by the not running infra controller. In fact this would fail https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/72f4dad21a8e72e7642d3e8327d9e4bbe14a19af/pkg/cloud/services/ec2/instances.go#L381